File: ios/Classes/Module/OrganizationManager.swift
import Foundation
import OpenIMCore

public class OrganizationManager: BaseServiceManager {

  public override func registerHandlers() {
        super.registerHandlers()
        self["setOrganizationListener"] = setOrganizationListener
        self["getSubDepartment"] = getSubDepartment
        self["getDepartmentMember"] = getDepartmentMember
        self["getUserInDepartment"] = getUserInDepartment
        self["getDepartmentMemberAndSubDepartment"] = getDepartmentMemberAndSubDepartment
        self["getDepartmentInfo"] = getDepartmentInfo
        self["searchOrganization"] = searchOrganization
    }

    func setOrganizationListener(methodCall: FlutterMethodCall, result: @escaping FlutterResult){
        Open_im_sdkSetOrganizationListener(OrganizationListener(channel: channel))
        callBack(result)
    }

    func getSubDepartment(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkGetSubDepartment(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "departmentID"], methodCall[int: "offset"], methodCall[int: "count"])
    }
    
    func getDepartmentMember(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkGetDepartmentMember(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "departmentID"], methodCall[int: "offset"], methodCall[int: "count"])
    }
    
    func getUserInDepartment(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkGetUserInDepartment(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "userID"])
    }
    
    func getDepartmentMemberAndSubDepartment(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkGetDepartmentMemberAndSubDepartment(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "departmentID"])
    }
    
    func getDepartmentInfo(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkGetDepartmentInfo(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "departmentID"])
    }
    
    func searchOrganization(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkSearchOrganization(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[jsonString: "searchParam"], methodCall[int: "offset"], methodCall[int: "count"])
    }
}

public class OrganizationListener: NSObject, Open_im_sdk_callbackOnOrganizationListenerProtocol {
    
    private let channel:FlutterMethodChannel

    init(channel:FlutterMethodChannel) {
        self.channel = channel
    }
    
    public func onOrganizationUpdated() {
        CommonUtil.emitEvent(channel: self.channel, method: "organizationListener", type: "onOrganizationUpdated", errCode: nil, errMsg: nil, data: nil)
    }
    
}

File: android/src/main/java/io/openim/flutter_openim_sdk/manager/OrganizationManager.java
package io.openim.flutter_openim_sdk.manager;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.openim.flutter_openim_sdk.listener.OnBaseListener;
import io.openim.flutter_openim_sdk.listener.OnOrganizationListener;
import open_im_sdk.Open_im_sdk;

public class OrganizationManager extends BaseManager {

    public void setOrganizationListener(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.setOrganizationListener(new OnOrganizationListener());
    }

    public void getSubDepartment(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.getSubDepartment(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "departmentID"),
                int2long(methodCall, "offset"),
                int2long(methodCall, "count")
        );
    }

    public void getDepartmentMember(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.getDepartmentMember(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "departmentID"),
                int2long(methodCall, "offset"),
                int2long(methodCall, "count")
        );
    }

    public void getUserInDepartment(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.getUserInDepartment(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "userID")
        );
    }

    public void getDepartmentMemberAndSubDepartment(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.getDepartmentMemberAndSubDepartment(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "departmentID")
        );
    }

    public void getDepartmentInfo(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.getDepartmentInfo(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "departmentID")
        );
    }

    public void searchOrganization(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.searchOrganization(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                jsonValue(methodCall, "searchParam"),
                int2long(methodCall, "offset"),
                int2long(methodCall, "count")
        );
    }
}
